logstash-core/src/main/java/org/logstash/instrument/witness/process/ProcessWitness.java

```diff
@@ -0,0 +1,247 @@
+package org.logstash.instrument.witness.process;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import com.sun.management.UnixOperatingSystemMXBean;
+import org.logstash.instrument.metrics.Metric;
+import org.logstash.instrument.metrics.gauge.NumberGauge;
+import org.logstash.instrument.witness.MetricSerializer;
+import org.logstash.instrument.witness.SerializableWitness;
+import org.logstash.instrument.witness.schedule.ScheduledWitness;
+
+import java.io.IOException;
+import java.lang.management.ManagementFactory;
+import java.lang.management.OperatingSystemMXBean;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A scheduled witness for process metrics
+ */
+@JsonSerialize(using = ProcessWitness.Serializer.class)
+public class ProcessWitness implements SerializableWitness, ScheduledWitness {
+
+    private static final OperatingSystemMXBean osMxBean = ManagementFactory.getOperatingSystemMXBean();
+    private final static String KEY = "process";
+    private static final Serializer SERIALIZER = new Serializer();
+    private final boolean isUnix;
+    private final NumberGauge openFileDescriptors;
+    private final NumberGauge peakOpenFileDescriptors;
+    private final NumberGauge maxFileDescriptors;
+    private final UnixOperatingSystemMXBean unixOsBean;
+    private final Cpu cpu;
+    private final Memory memory;
+    private final Snitch snitch;
+
+    /**
+     * Constructor
+     */
+    public ProcessWitness() {
+        this.openFileDescriptors = new NumberGauge("open_file_descriptors", -1);
+        this.maxFileDescriptors = new NumberGauge("max_file_descriptors", -1);
+        this.peakOpenFileDescriptors = new NumberGauge("peak_open_file_descriptors", -1);
+        this.isUnix = osMxBean instanceof UnixOperatingSystemMXBean;
```

Comment:
@jakelandis this is just a `static` imo, it will never change during the lifetime of the JVM :)

Reply:
will move to static final (or is final static ? ) ;)